### PR TITLE
feat: add some paths and utils

### PR DESCRIPTION
**Pull request by alexeagle**

### CODE REVIEW

File: lib/utils.bzl [hunk 1]
```diff
@@ -0,0 +1,10 @@
+"Public API"
+
+load("//lib/private:utils.bzl", "utils")
+
+is_external_label = utils.is_external_label
```
Comment (alexeagle) [inline lib/utils.bzl:5]: do you think we should decompose the struct here? or just re-export the whole struct?

**Review by mattem — APPROVED** [review]

File: lib/utils.bzl (hunk 1 above)
Comment (gregmagolan) [reply lib/utils.bzl:5]: I think people expect to import individual functions from a utils.bzl and not a struct of them

File: lib/private/utils.bzl [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        param: a string or label
+
+    Returns:
+        a bool
+    """
+    return len(_to_label(param).workspace_root) > 0
+
+# Path to the root of the monorepo
+def _path_to_root():
+    """ Retuns the path to the monorepo root under bazel
```
Comment (gregmagolan) [inline lib/private/utils.bzl:81]: monorepo => workspace

File: lib/private/utils.bzl [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    Returns:
+        Path to the monorepo root
```
Comment (gregmagolan) [inline lib/private/utils.bzl:84]: monorepo => workspace

File: lib/private/utils.bzl [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        "block-network",
+        "requires-fakeroot",
+        "exclusive",
+        "manual",
+        "external",
+    ]
+
+    return [tag for tag in tags if tag in WELL_KNOWN_TAGS]
+
+def _is_darwin(rctx):
```
Comment (gregmagolan) [inline lib/private/utils.bzl:30]: suppose we should include `_is_windows` and `_is_linux`?

**Review by gregmagolan — APPROVED** [review]
🌮

File: lib/private/paths.bzl [hunk 5]
```diff
@@ -0,0 +1,47 @@
+"""Path utils built on top of Skylib's path utils"""
+
+load("@bazel_skylib//lib:paths.bzl", _spaths = "paths")
+
+def _relative_file(to_file, frm_file):
```
Comment (gregmagolan) [inline lib/private/paths.bzl:5]: Would these params be better named `to` and `from` instead of `to_file` and `frm_file`?

Comment (alexeagle) [reply lib/private/paths.bzl:5]: from is a reserved keyword

**Pull request merged**